src/backend/columnar/columnar_customscan.c

```diff
@@ -137,18 +139,6 @@ columnar_customscan_init()
 }
 
 
-static void
-clear_paths(RelOptInfo *rel)
-{
-	rel->pathlist = NIL;
-	rel->partial_pathlist = NIL;
-	rel->cheapest_startup_path = NULL;
-	rel->cheapest_total_path = NULL;
-	rel->cheapest_unique_path = NULL;
-	rel->cheapest_parameterized_paths = NIL;
```

Comment:
Is it OK to leave these untouched? Could one of these reference a sequential scan on columnar?

Reply:
umm, looks like I should re-consider this, thanks for pointing out

Reply:
Thinking about this a bit more and realized that we should not just leave them untouched. For this reason, maybe I should do something like that:
* remove non-index paths from `pathlist`
* re-calculate `cheapest_startup_path`, `cheapest_total_path`, `cheapest_unique_path` & `cheapest_parameterized_paths` by simply iterating given `pathlist`

Does that sound good ?

Reply:
I think I found a simple solution for that. Calling `set_cheapest` function after removing non-index paths seems to be enough.

Reply:
Looks like `set_rel_pathlist` already calls `set_cheapest` after calling  `ColumnarSetRelPathlistHook` via `set_rel_pathlist_hook`, so I think we don't need to do anything.

Reply:
(let's continue discussing that from other review comment
https://github.com/citusdata/citus/pull/4950#discussion_r644108300)